modules/3d/include/opencv2/3d/volume_settings.hpp

```diff
@@ -0,0 +1,71 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+
+
+#ifndef OPENCV_3D_VOLUME_SETTINGS_HPP
+#define OPENCV_3D_VOLUME_SETTINGS_HPP
+
+#include <opencv2/core/cvstd.hpp>
+#include <opencv2/3d/volume.hpp>
+
+namespace cv
+{
+
+enum class VolumeType
+{
+    TSDF = 0,
+    HashTSDF = 1,
+    ColorTSDF = 2
+};
+
+
+class CV_EXPORTS_W VolumeSettings
+{
+public:
+    VolumeSettings();
+    VolumeSettings(VolumeType volumeType);
+    ~VolumeSettings();
+
+    void  setWidth(int val);
+    int   getWidth() const;
+    void  setHeight(int val);
+    int   getHeight() const;
+    void  setDepthFactor(float val);
+    float getDepthFactor() const;
+    void  setVoxelSize(float val);
+    float getVoxelSize() const;
+    void  setTruncatedDistance(float val);
+    float getTruncatedDistance() const;
+    void  setTruncateThreshold(float val);
+    float getTruncateThreshold() const;
```

Comment:
Today I found an issue which exists for a year+: `truncatedDistance` and `truncateThreshold` have similar names and therefore can be mixed in code. Now it's a good time to fix this.
As far as I understand:
* `truncatedDistance` is a scale for TSDF values
* `truncateThreshold` is a maximum distance for depth integrating input/raycasting output.
Therefore I propose you to rename `truncatedDistance`  to `tsdfScale` and `truncateThreshold` to `maxDepth` everywhere in the code.